packages/fluentui/react-bindings/src/FocusZone/FocusZone.tsx

```diff
@@ -58,9 +76,9 @@ const ALLOW_VIRTUAL_ELEMENTS = false;
 /**
  * Handle global tab presses so that we can patch tabindexes on the fly.
  */
-function _onKeyDownCapture(ev: KeyboardEvent) {
+function _onKeyDownCapture(this: Window, ev: KeyboardEvent) {
```

Comment:
Please don't call a function argument `this`, `window`, or any other keyword that means something special in javascript. It can throw people off.

Reply:
In this case is needed as it is part of the signature for the listener 

```typescript
listener: (this: Window, ev: WindowEventMap[K]) => any
```

Reply:
right but having the variables called "foo" doesn't make this behave differently.

Reply:
```jsx
const outZones = {
  _outerZones: new Map<Window, Set<FocusZone>>(),
  register(foo: Window, FZ: FocusZone) {
    if (this._outerZones.get(foo)) {
      this._outerZones.get(foo)?.add(FZ);
    } else {
      this._outerZones.set(foo, new Set([FZ]));
    }
  },
  unregister(foo: Window, FZ: FocusZone) {
    this._outerZones.get(foo)?.delete(FZ);
  },
  deleteOutZone(foo: Window) {
    this._outerZones.delete(foo);
  },
  getOutZone(foo: Window) {
    return this._outerZones.get(foo);
  },
};
```

Reply:
Actually it does, if we "rename" the `this` we get some errors when adding the listener, trying with `currentWindow` as Shift suggested for example:

![Screenshot 2020-08-18 at 16 54 38](https://user-images.githubusercontent.com/8545105/90529017-86f65280-e173-11ea-9154-a702c1e53139.png)

Reply:
oh I see what youre saying, but let me suggest a better alternative. Don't take this argument in. Leave the signature intact because that's the expected event hander arg (ev first).

To get the window, use the `getWindow` helper based on the ev.target element. I'm not sure if there is one in react-bindings, but there is one in the utilities package that you can look at. You basically need to derive the correct window based on the target that the event was triggered from.

Reply:
As the code stands right now, I believe it's compiling by accident. If you break into this code, you'll probably find the first argument (`this`) is actually the event args.

Reply:
I might be missing something though!

Reply:
Great idea, thanks @dzearing 

Solved with `getWindow`

```javascript
outerZones.getOutZone(getWindow(ev.target as Element)!)?.forEach(zone => zone.updateTabIndexes());
```